networkx/generators/trees.py

```diff
@@ -372,3 +418,730 @@ def random_tree(n, seed=None, create_using=None):
         tree.add_edges_from(edges)
 
     return tree
+
+
+@py_random_state("seed")
+def random_labeled_tree(n, *, seed=None):
+    """Returns a labeled tree on `n` nodes chosen uniformly at random.
+
+    Generating uniformly distributed random Prüfer sequences and
+    converting them into the corresponding trees is a straightforward
+    method of generating uniformly distributed random labeled trees.
+    This function implements this method.
+
+    Parameters
+    ----------
+    n : int
+        The number of nodes, greater than zero.
+    seed : random_state
+        Indicator of random number generation state.
+        See :ref:`Randomness<randomness>`
+
+    Returns
+    -------
+     :class:`networkx.Graph`
+        A `networkx.Graph` with nodes in the set {0, …, *n* - 1}.
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If `n` is zero (because the null graph is not a tree).
+    """
+    # Cannot create a Prüfer sequence unless `n` is at least two.
+    if n == 0:
+        raise nx.NetworkXPointlessConcept("the null graph is not a tree")
+    if n == 1:
+        return nx.empty_graph(1)
+    else:
+        sequence = [seed.choice(range(n)) for i in range(n - 2)]
+        return nx.from_prufer_sequence(sequence)
+
+
+@py_random_state("seed")
+def random_labeled_rooted_tree(n, *, seed=None):
```

Comment:
I'm not crazy about expanding the API for helpers like this, especially since there are already >20 functions in the `nx` namespace that start with `random_`, which IMO is a major hit to discoverability of *all* random functions. OTOH, I see the argument for symmetry with the unlabeled case. Not a blocker, but something to keep in mind when expanding this `random_` API surface!

Reply:
I understand your concerns. There was previous a version (see the conversation) in which labeling was selected via a parameter, but in turn it looked like people would not discover the importance of the difference between labeled and unlabeled.

Reply:
What do people think about `nx.labeled_rooted_random_tree` rather than `nx.random_labeled_rooted_tree`. It switches the adjective order -- promoting "labeled"/"unlabeled" to the first position.  It's almost like making `random_tree` be the noun with the qualifiers ahead of that. 

Dropping the grammar discussion -- it relieves the `random_` API surface.  Thoughts?

Reply:
> What do people think about nx.labeled_rooted_random_tree rather than nx.random_labeled_rooted_tree

This works! Though I expect "random tree" is a more well-known/searched-for concept than "labeled/unlabeled" random trees, so users doing `nx.r<tab>` will likely miss it.

Another option might be to expose a `random_trees` namespace with all of the functions contained therein rather than exporting each function to the top-level namespace individually. Then the user-access pattern would be something like: `nx.random_trees.labeled_rooted_tree` or `nx.random_trees.unlabeled_rooted_forest`. The downside of course being that this kind of breaks with the typical networkx pattern of functions being generally available in the main namespace (not to mention that there's already a deprecated `random_tree` name).

All of these approaches have their relative merits and downsides and I don't think there's any one obvious choice, so I don't want to block/bikeshed on it too much - I'd be perfectly happy with the current naming scheme getting merged!